CHANGELOG_PENDING.md

```diff
@@ -2,6 +2,18 @@
 
 ## v0.34.21
 
+Release highlights include:
+- A new RPC configuration flag, `discard_abci_responses`, which, if enabled,
+  discards all ABCI responses except the latest one in order to reduce disk
+  space usage in the state store. Note that, if enabled, this means that the
+  `block_results` RPC endpoint will only be able to return the latest block.
```

Comment:
Actually if enabled `/block_results` will return an error saying that the node isn't persisting abci responses

Reply:
I wonder if adding the RPC part is confusing. Yes it's in the RPC section but that's because we didn't want to create a new section and wasn't sure where such a change would best fit so we landed it in RPC but really this is a storage engine config

Reply:
> Actually if enabled `/block_results` will return an error saying that the node isn't persisting abci responses

Ah, I missed this somehow. In terms of the way it's implemented now though, does this make sense given that we do actually store the last ABCI response and can actually give the latest block results? Should we really always be returning an error if we have the results? (Mainly asking for context around the decision-making there)

> I wonder if adding the RPC part is confusing. Yes it's in the RPC section but that's because we didn't want to create a new section

Is it really a big deal to create a new section, especially when the flag's off by default? I'd say that, since it's technically a state store option, so it could either go into a `[statestore]` section, or more generically a `[storage]` section.

Reply:
> Ah, I missed this somehow. In terms of the way it's implemented now though, does this make sense given that we do actually store the last ABCI response and can actually give the latest block results? Should we really always be returning an error if we have the results? (Mainly asking for context around the decision-making there)

We definitely could. There needs to be some added logic but it's possible. The design decision was one of simplicity - easier just to close of the entire RPC endpoint than to have it error for every height except the latest

Reply:
> Is it really a big deal to create a new section, especially when the flag's off by default? I'd say that, since it's technically a state store option, so it could either go into a [statestore] section, or more generically a [storage] section.

It's possible to maybe have a general storage section. It's not clear to me yet what other configuration options there might be

Reply:
> We definitely could. There needs to be some added logic but it's possible.

Speaking to the IBC-rs team now, it may actually be better to just keep it as-is. The relayer could then just ping the `/block_results` endpoint on startup and, if it errors out with the current error, they can notify the user that the relevant node isn't correctly configured.